app/controllers/comments_controller.rb

```diff
@@ -40,10 +50,12 @@ def update
   # DELETE /comments/1
   # DELETE /comments/1.json
   def destroy
+    reference = @comment.root.commentable
```

Comment:
might be worth doing a before_action for `reference = @comment.root.commentable`